### DISCUSSION TITLE
Problemi di accesso a Staff ( avvisi ) dopo aggiornamento alla versione 1.5.1

### DISCUSSION THREAD

**Discussion by Forrest-It** [iisgiua/giuaschool#424] [Richieste di aiuto]
Dopo l'aggiornamento dalla versione 1.5.0 alla versione 1.5.1 nel ruolo di Staff all'accesso ho un errore di accesso alla pagina ( The server returned a "500 Internal Server Error". ) , segue parte del log relativo all'accesso :

[2022-11-17T17:16:31.801436+01:00] request.CRITICAL: Uncaught PHP Exception Error: "Call to a member function getId() on null" at src/Util/BachecaUtil.php line 162 {"exception":"[object] (Error(code: 0): Call to a member function getId() on null at src/Util/BachecaUtil.php:162)"} {"client_ip":"XXX","uri":"/staff/avvisi","query_string":null,"method":"GET","username":"XXX","roles":["ROLE_STAFF","ROLE_DOCENTE","ROLE_UTENTE"]}

E stato fatto sia l'aggiornamento alla versione 1.5.1 che l'ultimo aggiornamento della versione 1.5.1 : in entrambi i casi fatti manualmente .

Ho l'impressione che il problema sia legato al DB, a qualche campo mancante in qualche tabella ( non creato durante l'upgrade ):  utilizzando, infatti, lo stesso DB su una nuova installazione il problema rimane lo stesso .

Grazie

**Comment by trinko** [comment]
Si, l'errore aembra dovuto ai dati di un avviso che risulta non avere il riferimento al docente che l'ha inserito.
Puoi controllare la tabella **gs_avviso** e vedere se risulta qualche riga con il campo **docente_id** nullo.

**Comment by Forrest-It** [comment]
No, non ci sono record con docende_id nullo, ma ci sono record con materia_id e cattedra_id nullo. E' possibile che ci sia qualche problema con gli indici ?

![image](https://user-images.githubusercontent.com/73697572/202518254-953da387-d33b-4ae8-86d4-4b5fcdaa42cc.png)

**Comment by Forrest-It** [comment]
risolto : nel DB, non so come mai, c'erano tre avvisi associati a un id utente relativo a un alunno e a un genitore . Ho corretto l'avviso con il corretto id_docente e il problema è rientrato

**Comment by Forrest-It** [comment]
Mi invii a stretto giro i dati di accesso: ho una mezz'ora.

Saluti

Il Dom 20 Nov 2022, 14:35 gmontalbano5ct ***@***.***> ha
scritto:
